docs/integration_testing.md

```diff
@@ -0,0 +1,162 @@
+
+[pdi]:
+    https://developer.servicenow.com/dev.do#!/guides/quebec/developer-program/pdi-guide/personal-developer-instance-guide-introduction
+    (Introduction to Personal Developer Instances)
+[pdi-guide]:
+    https://developer.servicenow.com/dev.do#!/guides/quebec/developer-program/pdi-guide/obtaining-a-pdi
+    (Instructions on obtaining a Personal Developer Instance)
+
+## Running integration tests
+
+Now you are ready to run integration tests against your PDI:
+
+* To run all integration tests, use `make integration`.
+* To run integration tests for the specified target only, use `make <target>`,
+    for instance `make api_authentication`.
+
+## The general structure of integration tests
+
+Integration tests are grouped in _targets_. You will find them under
+`test/integration/targets` directory. Every test target is a role that will be
+executed when running `ansible-test integration`. A target should contain
+a `main.yml` file with your tests, i.e. this is where you put the tasks 
+that invoke the modules from the collection, and make assertions about the results.
+ 
+`main.yml` will typically contain many tasks using `servicenow.itsm`
+modules. Since all the modules require `instance` parameter, and
+all instance parameters can be specified via environment variables,
+we can improve readability and reduce lines of code by declaring
+a shared `environment` for all tasks, like this:
+    
+```yaml
+---
+# Environment for the block of tasks that follow.
+# Values of env vars are specified via the corresponding vars from integration_config.yml.
+- environment:
+    SN_HOST: "{{ sn_host }}"
+    SN_USERNAME: "{{ sn_username }}"
+    SN_PASSWORD: "{{ sn_password }}"
+    SN_CLIENT_ID: "{{ sn_client_id }}"
+    SN_CLIENT_SECRET: "{{ sn_client_secret }}"
+
+  block:
+    - servicenow.itsm.problem_info:
+        # No need to define instance parameter here, or for
+        # any of the subsequent tasks.
+      
+        # Other params relevant to your test
+        <other_module_params_here>
+      register: result
+    # Assertions about the result of the previous task
+    - ansible.builtin.assert:
+        that:
+           - result.records == []
+```
+
+Then, we can define our tests (i.e. tasks that invoke relevant modules from the
+collection, followed by assertions about their results) under `block`, 
+as illustrated in the example above.
+
+### Avoiding reliance on any pre-existing PDI data
+
+By default, Personal Developer Instances are populated with some initial data.
+To avoid failures due to PDIs being populated with different test data, integration
+tests should, wherever possible, avoid relying on any pre-existing data. 
+For instance, the assertions should avoid:
+
+* checking the total record count, or
+* checking the presence of a record that was not created by the module under test.  
+
+### Testing for expected module failures
+
+When we expect a module to fail, we specify `ignore_errors: True` for the task.
```

Comment:
Nitpick: should we go with `ignore_errors: true` here just to be consistent with examples?